container/runtime/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java

```diff
@@ -69,13 +71,34 @@ public void debug(boolean debug) {
     @Override
     public void deploy(Archive<?> deployment) throws DeploymentException {
 
+        // 1. give the factions a chance to handle the deployment
```

Comment:
Apologies Heiko, you're going to be annoyed by this! After saying what I did on Friday, I think, I realized that maybe some calls to prepareArchive() need to rely on things in the index? I'm not entirely sure, but it was something that popped into my head.

For instance, would we want prepareArchive() to be able to search for annotations, etc to then add/remove code from the Archive depending on what they find?

If so, then we do need to index first, and now you hate me ;-)

So, if we took that approach, do we need to re-index the index after prepareArchive()?